homeassistant/components/aftership/sensor.py

```diff
@@ -56,7 +78,40 @@
                       aftership.meta)
         return
 
-    async_add_entities([AfterShipSensor(aftership, name)], True)
+    instance = AfterShipSensor(aftership, name)
+
+    async_add_entities([instance], True)
+
+    async def handle_add_tracking(call):
+        """Call when a user adds a new Aftership tracking from HASS."""
+        title = call.data.get(CONF_TITLE)
+        slug = call.data.get(CONF_SLUG)
+        tracking_number = call.data[CONF_TRACKING_NUMBER]
+
+        await aftership.add_package_tracking(tracking_number, title, slug)
+        await instance.async_update(no_throttle=True)
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_ADD_TRACKING,
+        handle_add_tracking,
+        schema=ADD_TRACKING_SERVICE_SCHEMA,
+    )
+
+    async def handle_remove_tracking(call):
+        """Call when a user removes an Aftership tracking from HASS."""
+        slug = call.data[CONF_SLUG]
+        tracking_number = call.data[CONF_TRACKING_NUMBER]
+
+        await aftership.remove_package_tracking(slug, tracking_number)
+        await instance.async_update(no_throttle=True)
```

Comment:
Remove this and instead use `async_dispatcher_send` dispatch helper function to signal the entity to update.

Reply:
Is there an advantage to using the dispatcher or more a standard?

Reply:
The dispatcher allows separating the caller from the target function. This is very useful in some cases. The signal can be sent before the target function is connected without error. Multiple target functions can be connected to the same signal.

We use it as a standard when the component needs to call entity methods in platforms.

It can be used in other cases too when the features of the dispatcher fits.